### PR TITLE
[feature-wip](array-type) add agg function collect_list and collect_set

### PR DESCRIPTION
**Pull request by cambyzju**
# Proposed changes

Issue Number: close #7570

## Problem Summary:

`collect_list`
Returns an array consisting of all values in expr within the group.
The order of elements in the array is non-deterministic. NULL values are excluded.

`collect_set`
Returns an array consisting of all unique values in expr within the group.
The order of elements in the array is non-deterministic. NULL values are excluded.

Usage example:
```
mysql> set enable_vectorized_engine=true;
mysql> set enable_array_type = true;

mysql> select k1,k2,k3 from collect_test order by k1;
+------+------------+-------+
| k1   | k2         | k3    |
+------+------------+-------+
|    1 | 2022-07-05 | hello |
|    2 | 2022-07-04 | NULL  |
|    2 | 2022-07-04 | hello |
|    3 | NULL       | world |
|    3 | NULL       | world |
+------+------------+-------+

mysql> select k1,collect_list(k2),collect_list(k3) from collect_test group by k1 order by k1;
+------+--------------------------+--------------------+
| k1   | collect_list(`k2`)       | collect_list(`k3`) |
+------+--------------------------+--------------------+
|    1 | [2022-07-05]             | [hello]            |
|    2 | [2022-07-04, 2022-07-04] | [hello]            |
|    3 | NULL                     | [world, world]     |
+------+--------------------------+--------------------+

mysql> select k1,collect_set(k2),collect_set(k3) from collect_test group by k1 order by k1;
+------+-------------------+-------------------+
| k1   | collect_set(`k2`) | collect_set(`k3`) |
+------+-------------------+-------------------+
|    1 | [2022-07-05]      | [hello]           |
|    2 | [2022-07-04]      | [hello]           |
|    3 | NULL              | [world]           |
+------+-------------------+-------------------+
```

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/zh-CN/docs/sql-manual/sql-functions/aggregate-functions/collect_list.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+under the License.
+-->
+
+## COLLECT_LIST
+### description
+#### Syntax
+
+`ARRAY<T> collect_list(expr)`
+
+返回一个包含 epxr 中所有元素(不包括NULL)的数组，数组中元素顺序是不确定的。
```
Comment (BiteTheDDDDt) [inline docs/zh-CN/docs/sql-manual/sql-functions/aggregate-functions/collect_list.md:33]: `expr`

Comment (cambyzju) [reply docs/zh-CN/docs/sql-manual/sql-functions/aggregate-functions/collect_list.md]: done

**Review by BiteTheDDDDt — DISMISSED** [review]
LGTM

**Comment by cambyzju** [comment]
update regression-test output, because the output format for type ARRAY<String> have changed in pr10498.

https://github.com/apache/doris/pull/10498

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**